sockets/delegateMonitor.js

```diff
@@ -130,54 +168,123 @@ module.exports = function (app, connectionHandler, socket) {
         );
     };
 
-    var getLastBlocks = function (results) {
+    var getNextForgers = function (cb) {
+        if (running.getNextForgers) {
+            return cb('getNextForgers (already running)');
+        }
+        running.getNextForgers = true;
+        delegates.getNextForgers(
+            function (res) { running.getNextForgers = false; cb('NextForgers'); },
+            function (res) { running.getNextForgers = false; cb(null, res); }
+        );
+    };
+
+    var getLastBlocks = function (results, init) {
```

Comment:
That function looks sad - there is some callback hell there. It works ofc, but should be reviewed.